protocol.html

```diff
@@ -717,6 +719,51 @@ <h4 property="schema:name"><span>Note</span>: Conditional Update</h4>
+                      <li>The <code>?insertions</code> and <code>?deletions</code> formulae must not contain variables that do not occur in the <code>?conditions</code> formula.</li>
+                      <li>The <code>?insertions</code> and <code>?deletions</code> formulae must not contain blank nodes.</li>
+                      <li>The patch document does not contain any other triples.</li>
+                    </ul>
+
+                    <p><span about="" id="server-patch-n3-invalid" rel="spec:requirement" resource="#server-patch-n3-invalid"><span property="spec:statement"><span rel="spec:requirementSubject" resource="spec:Server">Servers</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> respond with a <code>400</code> status code if a patch document does not satisfy all of the above constraints.</span></span></p>
+
+                    <p><span about="" id="server-patch-n3-semantics" rel="spec:requirement" resource="#server-patch-n3-semantics"><span property="spec:statement"><span rel="spec:requirementSubject" resource="spec:Server">Servers</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> process a patch resource against the RDF dataset in the target document as follows, creating an empty RDF dataset if the target resource does not exist yet. If <code>?conditions</code> is non-empty, then the dataset must contain exactly one match of its contents after variable substitution; if not, the <span rel="spec:requirementSubject" resource="spec:Server">server</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> respond with a <code>409</code> status code. The resulting (possibly empty) variable mapping must be propagated to the <code>?deletions</code> and <code>?insertions</code> formulae. Then, the resulting triples in <code>?deletions</code> must be removed from the RDF dataset if there is exactly one match; if not, the <span rel="spec:requirementSubject" resource="spec:Server">server</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> respond with a <code>409</code> status code. Finally, the resulting triples in <code>?insertions</code> must be added to the RDF dataset, where any blank nodes must result in newly created blank nodes. The combination of the deletions and insertions then forms the new resource state of the RDF document, and the server responds with the appropriate status code.</span></span></p>
+                  </div>
+
+                  <figure id="n3-patch-example" class="example listing" rel="schema:hasPart" resource="#n3-patch-example">
+                    <p class="example-h"><span>Example</span>: Applying an N3 patch.</p>
+                    <pre about="#n3-patch-example" property="schema:description" typeof="fabio:Script">
+<code>@prefix solid: &lt;http://www.w3.org/ns/solid/terms#&gt;.
+@prefix ex: &lt;http://www.example.org/terms#&gt;.
+
+&lt;&gt; a solid:Patch;
```

Comment:
Not sure that `<>` is a good idea. That means "this resource", and that would resolve to the target resource, right? So, that would result in a URI collision between the target resource and the patch document. I think a blank node is actually a reasonable thing to have there...

Reply:
Unless I'm misunderstanding badly, I believe `<>` here resolves to the patch document.

Reply:
We assumed indeed that `<>` is the patch document; when @timbl and myself designed this, we wanted the thing to describe itself (hence also the `solid:patches`). So we wanted to convey "I am a patch that inserts/deletes …".
But now that a patch document can contain multiple patches, we probably want a different identifier altogether. Replaced in cb80918902b2e546f0313766db083f5919fb9520.